fix: improve error handling and key generation in dashboard components

MarvinFos opened the pull request:
Replace console.error with toast notifications for better user feedback Use titular.id as key with fallback to index in HistoryOwnership

MarvinFos committed:
fix: improve error handling and key generation in dashboard components

Replace console.error with toast notifications for better user feedback
Use titular.id as key with fallback to index in HistoryOwnership

EstebanFOS commented on src/modulos/DashDptos/UnitInfo/UnitInfo.tsx:
```diff
@@ -117,10 +119,10 @@ const UnitInfo = ({
+        showToast(data?.message || 'Error al cambiar titular', 'error');
       }
-    } catch (error) {
-      console.error('Error al llamar a /dptos-change-titular', error);
```
El mensaje secundario no  deberia de aparecer 'Error al llamar a /dptos-change-titular'

MarvinFos committed:
fix(DashDptos): improve error message when changing titular fails

Make error message more user-friendly by suggesting to contact admin when titular change fails

EstebanFOS commented:
## CODE REVIEW REPORT - Github Projects
**PR:** # [PR](https://github.com/mariogfos/condaty2025-admin/pull/267) - [Github Dashboard issue](https://github.com/Condaty/admin/issues/21)
**App:** Admin  
**Autor:** @MarvinFos 
**Revisor:** @EstebanFOS 
**Fecha:** 05/12/25  
**Estado:**  **APROBADO**

---

## PROBLEMAS CRÍTICOS (BLOCKERS)
- [x] **Seguridad:** Todo correcto
- [x] **Funcionalidad rota:** No se detectan problemas
- [x] **Performance crítica:** Sin incidencias

---

## ISSUES A CORREGIR
### **Alta Prioridad** (Merge bloqueado)
- [x] Ningún issue detectado

### **Media Prioridad** (Corregir antes de deploy)
- [x] Ningún issue detectado

### **Baja Prioridad** (Mejora futura)
- [x] Ninguna sugerencia pendiente

---

## CHECKLIST RÁPIDO (Solo marcar problemas)
**Calidad de código:**
- [x] Nombres claros
- [x] Sin código duplicado
- [x] Lógica sencilla

**Front específico:**
- [x] Sin props drilling excesivo
- [x] Sin re-renderizados innecesarios
- [x] Accesibilidad correcta

**Seguridad:**
- [x] Inputs validados
- [x] Sin tokens/keys expuestos
- [x] Permisos correctos

**Tests:**
- [x] Tests completos para nueva lógica
- [x] Coverage estable

---

## DECISIÓN FINAL
**Veredicto:** APROBADO  
**Acciones inmediatas:**  
- [x] No se requieren correcciones  
- [x] Listo para merge
